p4proto/bfIntf/bf_chassis_manager.h

```diff
@@ -25,10 +25,29 @@
 #define GNMI_CONFIG_QUEUE_COUNT 0x04
 #define GNMI_CONFIG_SOCKET_PATH 0x08
 #define GNMI_CONFIG_HOST_NAME 0x10
+#define GNMI_CONFIG_PIPELINE_NAME 0x20
+#define GNMI_CONFIG_MEMPOOL_NAME 0x40
+#define GNMI_CONFIG_MTU_VALUE 0x80
 
-#define GNMI_CONFIG_TDI (GNMI_CONFIG_PORT_TYPE | GNMI_CONFIG_DEVICE_TYPE | \
-                         GNMI_CONFIG_QUEUE_COUNT | GNMI_CONFIG_SOCKET_PATH | \
-                         GNMI_CONFIG_HOST_NAME)
+#define GNMI_CONFIG_VHOST (GNMI_CONFIG_PORT_TYPE | GNMI_CONFIG_DEVICE_TYPE | \
+                           GNMI_CONFIG_QUEUE_COUNT | GNMI_CONFIG_SOCKET_PATH | \
+                           GNMI_CONFIG_HOST_NAME)
+
+
+#define GNMI_CONFIG_TAP (GNMI_CONFIG_PORT_TYPE)
+
+// Independant TAP ports shouldn't have the below params.
+#define GNMI_CONFIG_UNSUPPORTED_MASK_TAP (GNMI_CONFIG_DEVICE_TYPE | GNMI_CONFIG_QUEUE_COUNT | \
+                                          GNMI_CONFIG_SOCKET_PATH | GNMI_CONFIG_HOST_NAME)
+
```

Comment:
Why use a negative mask instead of defining a seperate GNMI_CONFIG_TAP mask defining the TAP parameters? That will make it easier to add other link types later on with other specific parameters if needed ?